commit 6df23f4c5cef5b6d8879720ef7c842adffc565c9
Author: Dennis Li <Dennis.Li@amd.com>
Date:   Fri Apr 16 22:41:11 2021 +0800

    drm/amdgpu: fix a error injection failed issue
    
    because "sscanf(str, "retire_page")" always return 0, if application use
    the raw data for error injection, it always wrongly falls into "op ==
    3". Change to use strstr instead.
    
    Signed-off-by: Dennis Li <Dennis.Li@amd.com>
    Reviewed-by: Hawking Zhang <Hawking.Zhang@amd.com>
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c
index 38a691a3b600..7438d4e84776 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_ras.c
@@ -221,7 +221,7 @@ static int amdgpu_ras_debugfs_ctrl_parse_data(struct file *f,
 		op = 1;
 	else if (sscanf(str, "inject %32s %8s", block_name, err) == 2)
 		op = 2;
-	else if (sscanf(str, "retire_page") == 0)
+	else if (strstr(str, "retire_page") != NULL)
 		op = 3;
 	else if (str[0] && str[1] && str[2] && str[3])
 		/* ascii string, but commands are not matched. */